internal/components/setup/kind.go

```diff
@@ -0,0 +1,92 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package setup
+
+import (
+	"bytes"
+	"os/exec"
+	"strings"
+
+	"github.com/apache/skywalking-infra-e2e/internal/flags"
+	"github.com/apache/skywalking-infra-e2e/internal/logger"
+)
+
+type ExecResult struct {
+	Command []string
+	Error   error
+	Stdout  string
+	StdErr  string
+}
+
+const (
+	KIND        = "kind"
+	KINDCOMMAND = "kind"
+)
+
+var (
+	// kind cluster create config
+	kindConfigFile string
+)
+
+// setup for kind, invoke from command line
+func KindSetupInCommand() {
+	kindConfigFile = flags.File
+
+	execResult := createKindCluster()
+	err := execResult.Error
+	if err != nil {
+		cmd := strings.Join(execResult.Command, " ")
+		logger.Log.Errorf("Kind cluster create exited abnormally whilst running [%s]\n"+
+			"err: %s\nstdout: %s\nstderr: %s", cmd, err, execResult.Stdout, execResult.StdErr)
+	} else {
+		defer cleanupKindCluster()
```

Comment:
The cleanup step is not supposed to be here (at the end of `setup`), instead, it should be at the end of the `run` command (if the `run` command is executed), otherwise if we only execute `setup`, no `cleanup` should be done because we are doing step by step debugging.

Reply:
i see